src/main/errors.c

```diff
@@ -3006,3 +3006,40 @@ void R_InitConditions(void)
     MARK_NOT_MUTABLE(R_nodeStackOverflowError);
     R_PreserveObject(R_nodeStackOverflowError);
 }
+attribute_hidden SEXP R_makeCTimeLimitError(SEXP call,double value, const char  *classname, const char *fieldname ){
+	SEXP cond = PROTECT( R_makeErrorCondition(call, 
+		classname ,
+		"TimeLimitError",
+		1,
+		C_TimeLimit_msg_fmt,
+        value 
+	)) ; 
+	R_setConditionField(cond, 2, fieldname, ScalarReal(value));
+
+	UNPROTECT(1);
+    return cond;
+}
+
+attribute_hidden
+SEXP R_makeElapsedTimeLimitError(SEXP call,double elapsed )
+{
+       return R_makeTimeLimitError(call,
+								   elapsed,
+								   "elapsed_TimeLimitError",
+								   "elapsed");
+}
+
+attribute_hidden
+SEXP R_makeCpuTimeLimitError(SEXP call, double cpu_time)
+{
+    return R_makeTimeLimitError(call,
+                                cpu_time,
+                                "cpu_TimeLimitError",
+                                "cpu");
+} 
+
```

Comment:
I am not sure how much by R Core you have looked at but I do not think they would be thrilled with this indentation default....